Build failure: cargo-watch (x86-64-darwin)

### Steps To Reproduce

Steps to reproduce the behavior:
1. build cargo-watch on x86-64-darwin

### Build log

```
unpacking sources
unpacking source archive /nix/store/wsla3x0rznrxfslm0qhw4bnndwkllwll-source
source root is source
Executing cargoSetupPostUnpackHook
unpacking source archive /nix/store/1sf1ahan4iid05c9ppw57ia71bk8pc3s-cargo-watch-8.4.1-vendor.tar.gz
Finished cargoSetupPostUnpackHook
patching sources
Executing cargoSetupPostPatchHook
Validating consistency between /private/tmp/nix-build-cargo-watch-8.4.1.drv-0/source/Cargo.lock and /private/tmp/nix-build-cargo-watch-8.4.1.drv-0/cargo-watch-8.4.1-vendor.tar.gz/Cargo.lock
Finished cargoSetupPostPatchHook
configuring
building
Executing cargoBuildHook
++ env CC_X86_64_APPLE_DARWIN=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc CXX_X86_64_APPLE_DARWIN=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/c++ CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc CC_X86_64_APPLE_DARWIN=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc CXX_X86_64_APPLE_DARWIN=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/c++ CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc CARGO_BUILD_TARGET=x86_64-apple-darwin HOST_CC=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc HOST_CXX=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/c++ cargo build -j 2 --target x86_64-apple-darwin --frozen --profile release
   Compiling libc v0.2.147
   Compiling proc-macro2 v1.0.66
   Compiling unicode-ident v1.0.11
   Compiling quote v1.0.33
   Compiling cfg-if v1.0.0
   Compiling syn v2.0.29
   Compiling autocfg v1.1.0
   Compiling memchr v2.6.1
   Compiling syn v1.0.109
   Compiling rand_core v0.6.4
   Compiling siphasher v0.3.11
   Compiling bitflags v1.3.2
   Compiling phf_shared v0.11.2
   Compiling rand v0.8.5
   Compiling serde v1.0.188
   Compiling strsim v0.10.0
   Compiling ident_case v1.0.1
   Compiling fnv v1.0.7
   Compiling phf_generator v0.11.2
   Compiling serde_derive v1.0.188
   Compiling darling_core v0.12.4
   Compiling version_check v0.9.4
   Compiling nom v5.1.3
   Compiling darling_macro v0.12.4
   Compiling phf_codegen v0.11.2
   Compiling memoffset v0.6.5
   Compiling malloc_buf v0.0.6
   Compiling thiserror v1.0.47
   Compiling objc v0.2.7
   Compiling cc v1.0.83
   Compiling terminfo v0.7.5
   Compiling darling v0.12.4
   Compiling thiserror-impl v1.0.47
   Compiling aho-corasick v1.0.5
   Compiling num-traits v0.2.16
   Compiling dirs-sys v0.3.7
   Compiling regex-syntax v0.7.5
   Compiling regex-automata v0.3.7
   Compiling dirs v4.0.0
   Compiling mac-notification-sys v0.6.1
   Compiling derive_builder_core v0.10.2
   Compiling objc_id v0.1.1
   Compiling phf v0.11.2
   Compiling fsevent-sys v2.0.1
   Compiling dirs-sys-next v0.1.2
   Compiling serde_json v1.0.105
   Compiling block v0.1.6
   Compiling time-core v0.1.1
   Compiling core-foundation-sys v0.8.4
   Compiling camino v1.1.6
   Compiling log v0.4.20
   Compiling semver v1.0.18
   Compiling deranged v0.3.8
   Compiling either v1.9.0
   Compiling same-file v1.0.6
   Compiling walkdir v2.3.3
   Compiling time v0.3.28
   Compiling which v4.4.0
   Compiling iana-time-zone v0.1.57
   Compiling objc-foundation v0.1.1
   Compiling dirs-next v2.0.0
   Compiling fsevent v0.4.0
   Compiling nix v0.22.3
   Compiling derive_builder_macro v0.10.2
   Compiling regex v1.9.4
The following warnings were emitted during compilation:

warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'CoreFoundation' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:60:
warning: In file included from /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFPropertyList.h:17:
warning: In file included from /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:16:
warning: /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1189:83: error: expected ','
warning:     kCFURLBookmarkCreationWithSecurityScope API_AVAILABLE(macos(10.7), macCatalyst(13.0))  API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 11 ), // Mac OS X 10.7.3 and later, include information in the bookmark data which allows the same sandboxed process to access the resource after being relaunched
warning:                                                                                   ^
warning: /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1190:98: error: expected ','
warning:     kCFURLBookmarkCreationSecurityScopeAllowOnlyReadAccess API_AVAILABLE(macos(10.7), macCatalyst(13.0)) API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 12 ), // Mac OS X 10.7.3 and later, if used with kCFURLBookmarkCreationWithSecurityScope, at resolution time only read access to the resource will be granted
warning:                                                                                                  ^
warning: /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1200:85: error: expected ','
warning:     kCFURLBookmarkResolutionWithSecurityScope API_AVAILABLE(macos(10.7), macCatalyst(13.0)) API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 10 ), // Mac OS X 10.7.3 and later, use the secure information included at creation time to provide the ability to access the resource in a sandboxed process.
warning:                                                                                     ^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: In file included from <module-includes>:1:
warning: /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'CoreGraphics' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSGeometry.h:9:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/870aqamwwkdvq6jyvx1387x56j9j1c25-apple-framework-CoreGraphics-11.0.0/Library/Frameworks/CoreGraphics.framework/Headers/CoreGraphics.h:8:
warning: /nix/store/870aqamwwkdvq6jyvx1387x56j9j1c25-apple-framework-CoreGraphics-11.0.0/Library/Frameworks/CoreGraphics.framework/Headers/CGBase.h:12:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CFBase.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'CoreGraphics' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSGeometry.h:9:
warning: While building module 'IOKit' imported from /nix/store/870aqamwwkdvq6jyvx1387x56j9j1c25-apple-framework-CoreGraphics-11.0.0/Library/Frameworks/CoreGraphics.framework/Headers/CGDisplayConfiguration.h:12:
warning: In file included from <module-includes>:4:
warning: /nix/store/jx6ybwk1fz84ajfdng45xq65qqgpxjyg-apple-framework-IOKit-11.0.0/Library/Frameworks/IOKit.framework/Headers/IOCFBundle.h:26:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'Security' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURLCredential.h:9:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/0b4412s4mvqz5h9j8r1zighkkd1q8hln-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/Security.h:27:
warning: /nix/store/0b4412s4mvqz5h9j8r1zighkkd1q8hln-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:28:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CFBase.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'CoreServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:15:
warning: In file included from <module-includes>:1:
warning: /nix/store/q4f6dscms9mjj3fmsra20fjbwy30l0m6-apple-framework-CoreServices-11.0.0/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'CoreServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:15:
warning: While building module 'DiskArbitration' imported from /nix/store/q4f6dscms9mjj3fmsra20fjbwy30l0m6-apple-framework-CoreServices-11.0.0/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:56:
warning: In file included from <module-includes>:1:
warning: /nix/store/ymiy4nkbjkjzin6h9hn28q1j37cz2qs4-apple-framework-DiskArbitration-11.0.0/Library/Frameworks/DiskArbitration.framework/Headers/DiskArbitration.h:27:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'CoreServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:15:
warning: While building module 'CFNetwork' imported from /nix/store/q4f6dscms9mjj3fmsra20fjbwy30l0m6-apple-framework-CoreServices-11.0.0/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:31:
warning: In file included from <module-includes>:1:
warning: /nix/store/gib7vpyyxhayrmd1shz7awka3vhpdd4c-apple-framework-CFNetwork-11.0.0/Library/Frameworks/CFNetwork.framework/Headers/CFNetwork.h:18:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'ApplicationServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:9:
warning: In file included from <module-includes>:1:
warning: /nix/store/wfbygiic3lq2aa6gcwcxmx2fcf5xd3cl-apple-framework-ApplicationServices-11.0.0/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:10: fatal error: could not build module 'CoreServices'
warning: #include <CoreServices/CoreServices.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'ApplicationServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:9:
warning: While building module 'CoreText' imported from /nix/store/wfbygiic3lq2aa6gcwcxmx2fcf5xd3cl-apple-framework-ApplicationServices-11.0.0/Library/Frameworks/ApplicationServices.framework/Frameworks/ATS.framework/Headers/SFNTLayoutTypes.h:16:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/sw9j47hx0hjc53x5yrnkrmpwk6ksh8i5-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CoreText.h:20:
warning: In file included from /nix/store/sw9j47hx0hjc53x5yrnkrmpwk6ksh8i5-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CTFont.h:20:
warning: In file included from /nix/store/sw9j47hx0hjc53x5yrnkrmpwk6ksh8i5-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CTFontDescriptor.h:20:
warning: /nix/store/sw9j47hx0hjc53x5yrnkrmpwk6ksh8i5-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CTFontTraits.h:14:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'ApplicationServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:9:
warning: While building module 'ColorSync' imported from /nix/store/wfbygiic3lq2aa6gcwcxmx2fcf5xd3cl-apple-framework-ApplicationServices-11.0.0/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:31:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/z9lqylmlkwbb1fz0i0i5k187w7i4ym2z-apple-framework-ColorSync-11.0.0/Library/Frameworks/ColorSync.framework/Headers/ColorSync.h:10:
warning: /nix/store/z9lqylmlkwbb1fz0i0i5k187w7i4ym2z-apple-framework-ColorSync-11.0.0/Library/Frameworks/ColorSync.framework/Headers/ColorSyncBase.h:12:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
warning: While building module 'ApplicationServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:9:
warning: While building module 'ImageIO' imported from /nix/store/wfbygiic3lq2aa6gcwcxmx2fcf5xd3cl-apple-framework-ApplicationServices-11.0.0/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:47:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/qim7hs1kjfnkxwwzsdy7gj6ykb0k5xxy-apple-framework-ImageIO-11.0.0/Library/Frameworks/ImageIO.framework/Headers/ImageIO.h:10:
warning: /nix/store/qim7hs1kjfnkxwwzsdy7gj6ykb0k5xxy-apple-framework-ImageIO-11.0.0/Library/Frameworks/ImageIO.framework/Headers/ImageIOBase.h:16:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: In file included from <module-includes>:1:
warning: /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:9: fatal error: could not build module 'Foundation'
warning: #import <Foundation/Foundation.h>
warning:  ~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: In file included from <module-includes>:1:
warning: /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/AppKit.h:10:9: fatal error: could not build module 'Foundation'
warning: #import <Foundation/Foundation.h>
warning:  ~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: While building module 'CoreData' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSKeyValueBinding.h:12:
warning: In file included from <module-includes>:1:
warning: /nix/store/904ymlq8dn6q5hi9yaq6z77628jis6r3-apple-framework-CoreData-11.0.0/Library/Frameworks/CoreData.framework/Headers/CoreData.h:8:9: fatal error: could not build module 'Foundation'
warning: #import <Foundation/Foundation.h>
warning:  ~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: While building module 'CoreData' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSKeyValueBinding.h:12:
warning: While building module 'CloudKit' imported from /nix/store/904ymlq8dn6q5hi9yaq6z77628jis6r3-apple-framework-CoreData-11.0.0/Library/Frameworks/CoreData.framework/Headers/NSPersistentCloudKitContainerOptions.h:10:
warning: In file included from <module-includes>:1:
warning: /nix/store/19lp09hc36s4aijcyyszjb09ammi5ill-apple-framework-CloudKit-11.0.0/Library/Frameworks/CloudKit.framework/Headers/CloudKit.h:8:9: fatal error: could not build module 'Foundation'
warning: #import <Foundation/Foundation.h>
warning:  ~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: While building module 'CoreData' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSKeyValueBinding.h:12:
warning: While building module 'CloudKit' imported from /nix/store/904ymlq8dn6q5hi9yaq6z77628jis6r3-apple-framework-CoreData-11.0.0/Library/Frameworks/CoreData.framework/Headers/NSPersistentCloudKitContainerOptions.h:10:
warning: While building module 'CoreLocation' imported from /nix/store/19lp09hc36s4aijcyyszjb09ammi5ill-apple-framework-CloudKit-11.0.0/Library/Frameworks/CloudKit.framework/Headers/CKRecord.h:9:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/48ppfx34bv5nxzxj1wa0n521hj3nixsz-apple-framework-CoreLocation-11.0.0/Library/Frameworks/CoreLocation.framework/Headers/CoreLocation.h:18:
warning: /nix/store/48ppfx34bv5nxzxj1wa0n521hj3nixsz-apple-framework-CoreLocation-11.0.0/Library/Frameworks/CoreLocation.framework/Headers/CLErrorDomain.h:10:9: fatal error: could not build module 'Foundation'
warning: #import <Foundation/Foundation.h>
warning:  ~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: While building module 'CoreImage' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSColor.h:51:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CoreImage.h:8:
warning: /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CoreImageDefines.h:19:10: fatal error: could not build module 'ApplicationServices'
warning: #include <ApplicationServices/ApplicationServices.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: While building module 'CoreImage' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSColor.h:51:
warning: While building module 'CoreVideo' imported from /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CIImage.h:10:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CoreVideo.h:20:
warning: In file included from /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CVReturn.h:21:
warning: /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CVBase.h:28:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CFBase.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: While building module 'CoreImage' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSColor.h:51:
warning: While building module 'CoreVideo' imported from /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CIImage.h:10:
warning: While building module 'IOSurface' imported from /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CVPixelBufferIOSurface.h:31:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/d8an1vz0bmni8fz7smvwawi2wksbaa2g-apple-framework-IOSurface-11.0.0/Library/Frameworks/IOSurface.framework/Headers/IOSurface.h:12:
warning: /nix/store/d8an1vz0bmni8fz7smvwawi2wksbaa2g-apple-framework-IOSurface-11.0.0/Library/Frameworks/IOSurface.framework/Headers/IOSurfaceBase.h:36:10: fatal error: could not build module 'CoreFoundation'
warning: #include <CoreFoundation/CoreFoundation.h>
warning:  ~~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: While building module 'CoreImage' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSColor.h:51:
warning: While building module 'CoreVideo' imported from /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CIImage.h:10:
warning: While building module 'Metal' imported from /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CVMetalTextureCache.h:41:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/lbxzwnmn6lmmwzbdzlhw7fyq102ldqyh-apple-framework-Metal-11.0.0/Library/Frameworks/Metal.framework/Headers/Metal.h:9:
warning: /nix/store/lbxzwnmn6lmmwzbdzlhw7fyq102ldqyh-apple-framework-Metal-11.0.0/Library/Frameworks/Metal.framework/Headers/MTLTypes.h:8:9: fatal error: could not build module 'Foundation'
warning: #import <Foundation/Foundation.h>
warning:  ~~~~~~~^
warning: While building module 'Cocoa' imported from objc/notify.h:1:
warning: While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
warning: While building module 'QuartzCore' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSOpenGLLayer.h:11:
warning: In file included from <module-includes>:1:
warning: In file included from /nix/store/9699m64kz6cn19yfrypmcm7i72k25bhr-apple-framework-QuartzCore-11.0.0/Library/Frameworks/QuartzCore.framework/Headers/QuartzCore.h:9:
warning: /nix/store/9699m64kz6cn19yfrypmcm7i72k25bhr-apple-framework-QuartzCore-11.0.0/Library/Frameworks/QuartzCore.framework/Headers/CoreImage.h:1:10: fatal error: could not build module 'CoreImage'
warning: #include <CoreImage/CoreImage.h>
warning:  ~~~~~~~~^
warning: In file included from objc/notify.m:1:
warning: objc/notify.h:1:9: fatal error: could not build module 'Cocoa'
warning: #import <Cocoa/Cocoa.h>
warning:  ~~~~~~~^
warning: 25 errors generated.

error: failed to run custom build command for `mac-notification-sys v0.6.1`

Caused by:
  process didn't exit successfully: `/private/tmp/nix-build-cargo-watch-8.4.1.drv-0/source/target/release/build/mac-notification-sys-612a95320e38153d/build-script-build` (exit status: 1)
  --- stdout
  TARGET = Some("x86_64-apple-darwin")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-apple-darwin")
  cargo:rerun-if-env-changed=CC_x86_64-apple-darwin
  CC_x86_64-apple-darwin = None
  cargo:rerun-if-env-changed=CC_x86_64_apple_darwin
  CC_x86_64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = Some("/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("cmpxchg16b,fxsr,sse,sse2,sse3,ssse3")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-apple-darwin
  CFLAGS_x86_64-apple-darwin = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_apple_darwin
  CFLAGS_x86_64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  running: "/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-arch" "x86_64" "-Wall" "-Wextra" "-fmodules" "-Wno-deprecated-declarations" "-mmacos-version-min=10.12" "-o" "/private/tmp/nix-build-cargo-watch-8.4.1.drv-0/source/target/x86_64-apple-darwin/release/build/mac-notification-sys-d4149a46f10e6591/out/objc/notify.o" "-c" "objc/notify.m"
  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'CoreFoundation' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:60:

  cargo:warning=In file included from /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFPropertyList.h:17:

  cargo:warning=In file included from /nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:16:

  cargo:warning=/nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1189:83: error: expected ','

  cargo:warning=    kCFURLBookmarkCreationWithSecurityScope API_AVAILABLE(macos(10.7), macCatalyst(13.0))  API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 11 ), // Mac OS X 10.7.3 and later, include information in the bookmark data which allows the same sandboxed process to access the resource after being relaunched

  cargo:warning=                                                                                  ^

  cargo:warning=/nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1190:98: error: expected ','

  cargo:warning=    kCFURLBookmarkCreationSecurityScopeAllowOnlyReadAccess API_AVAILABLE(macos(10.7), macCatalyst(13.0)) API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 12 ), // Mac OS X 10.7.3 and later, if used with kCFURLBookmarkCreationWithSecurityScope, at resolution time only read access to the resource will be granted

  cargo:warning=                                                                                                 ^

  cargo:warning=/nix/store/8awvj7chyz4fg7gyz408ihb6z7nq2617-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1200:85: error: expected ','

  cargo:warning=    kCFURLBookmarkResolutionWithSecurityScope API_AVAILABLE(macos(10.7), macCatalyst(13.0)) API_UNAVAILABLE(ios, watchos, tvos) = ( 1UL << 10 ), // Mac OS X 10.7.3 and later, use the secure information included at creation time to provide the ability to access the resource in a sandboxed process.

  cargo:warning=                                                                                    ^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'CoreGraphics' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSGeometry.h:9:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/870aqamwwkdvq6jyvx1387x56j9j1c25-apple-framework-CoreGraphics-11.0.0/Library/Frameworks/CoreGraphics.framework/Headers/CoreGraphics.h:8:

  cargo:warning=/nix/store/870aqamwwkdvq6jyvx1387x56j9j1c25-apple-framework-CoreGraphics-11.0.0/Library/Frameworks/CoreGraphics.framework/Headers/CGBase.h:12:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CFBase.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'CoreGraphics' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSGeometry.h:9:

  cargo:warning=While building module 'IOKit' imported from /nix/store/870aqamwwkdvq6jyvx1387x56j9j1c25-apple-framework-CoreGraphics-11.0.0/Library/Frameworks/CoreGraphics.framework/Headers/CGDisplayConfiguration.h:12:

  cargo:warning=In file included from <module-includes>:4:

  cargo:warning=/nix/store/jx6ybwk1fz84ajfdng45xq65qqgpxjyg-apple-framework-IOKit-11.0.0/Library/Frameworks/IOKit.framework/Headers/IOCFBundle.h:26:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'Security' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURLCredential.h:9:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/0b4412s4mvqz5h9j8r1zighkkd1q8hln-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/Security.h:27:

  cargo:warning=/nix/store/0b4412s4mvqz5h9j8r1zighkkd1q8hln-apple-framework-Security-11.0.0/Library/Frameworks/Security.framework/Headers/SecBase.h:28:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CFBase.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'CoreServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:15:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/q4f6dscms9mjj3fmsra20fjbwy30l0m6-apple-framework-CoreServices-11.0.0/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'CoreServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:15:

  cargo:warning=While building module 'DiskArbitration' imported from /nix/store/q4f6dscms9mjj3fmsra20fjbwy30l0m6-apple-framework-CoreServices-11.0.0/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:56:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/ymiy4nkbjkjzin6h9hn28q1j37cz2qs4-apple-framework-DiskArbitration-11.0.0/Library/Frameworks/DiskArbitration.framework/Headers/DiskArbitration.h:27:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'CoreServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:15:

  cargo:warning=While building module 'CFNetwork' imported from /nix/store/q4f6dscms9mjj3fmsra20fjbwy30l0m6-apple-framework-CoreServices-11.0.0/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:31:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/gib7vpyyxhayrmd1shz7awka3vhpdd4c-apple-framework-CFNetwork-11.0.0/Library/Frameworks/CFNetwork.framework/Headers/CFNetwork.h:18:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'ApplicationServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:9:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/wfbygiic3lq2aa6gcwcxmx2fcf5xd3cl-apple-framework-ApplicationServices-11.0.0/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:10: fatal error: could not build module 'CoreServices'

  cargo:warning=#include <CoreServices/CoreServices.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'ApplicationServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:9:

  cargo:warning=While building module 'CoreText' imported from /nix/store/wfbygiic3lq2aa6gcwcxmx2fcf5xd3cl-apple-framework-ApplicationServices-11.0.0/Library/Frameworks/ApplicationServices.framework/Frameworks/ATS.framework/Headers/SFNTLayoutTypes.h:16:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/sw9j47hx0hjc53x5yrnkrmpwk6ksh8i5-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CoreText.h:20:

  cargo:warning=In file included from /nix/store/sw9j47hx0hjc53x5yrnkrmpwk6ksh8i5-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CTFont.h:20:

  cargo:warning=In file included from /nix/store/sw9j47hx0hjc53x5yrnkrmpwk6ksh8i5-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CTFontDescriptor.h:20:

  cargo:warning=/nix/store/sw9j47hx0hjc53x5yrnkrmpwk6ksh8i5-apple-framework-CoreText-11.0.0/Library/Frameworks/CoreText.framework/Headers/CTFontTraits.h:14:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'ApplicationServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:9:

  cargo:warning=While building module 'ColorSync' imported from /nix/store/wfbygiic3lq2aa6gcwcxmx2fcf5xd3cl-apple-framework-ApplicationServices-11.0.0/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:31:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/z9lqylmlkwbb1fz0i0i5k187w7i4ym2z-apple-framework-ColorSync-11.0.0/Library/Frameworks/ColorSync.framework/Headers/ColorSync.h:10:

  cargo:warning=/nix/store/z9lqylmlkwbb1fz0i0i5k187w7i4ym2z-apple-framework-ColorSync-11.0.0/Library/Frameworks/ColorSync.framework/Headers/ColorSyncBase.h:12:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'Foundation' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:

  cargo:warning=While building module 'ApplicationServices' imported from /nix/store/q0s7bd7aaph3h7dmdr2096s5c93g8hig-apple-framework-Foundation-11.0.0/Library/Frameworks/Foundation.framework/Headers/NSAppleEventDescriptor.h:9:

  cargo:warning=While building module 'ImageIO' imported from /nix/store/wfbygiic3lq2aa6gcwcxmx2fcf5xd3cl-apple-framework-ApplicationServices-11.0.0/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:47:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/qim7hs1kjfnkxwwzsdy7gj6ykb0k5xxy-apple-framework-ImageIO-11.0.0/Library/Frameworks/ImageIO.framework/Headers/ImageIO.h:10:

  cargo:warning=/nix/store/qim7hs1kjfnkxwwzsdy7gj6ykb0k5xxy-apple-framework-ImageIO-11.0.0/Library/Frameworks/ImageIO.framework/Headers/ImageIOBase.h:16:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:9: fatal error: could not build module 'Foundation'

  cargo:warning=#import <Foundation/Foundation.h>

  cargo:warning= ~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/AppKit.h:10:9: fatal error: could not build module 'Foundation'

  cargo:warning=#import <Foundation/Foundation.h>

  cargo:warning= ~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=While building module 'CoreData' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSKeyValueBinding.h:12:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/904ymlq8dn6q5hi9yaq6z77628jis6r3-apple-framework-CoreData-11.0.0/Library/Frameworks/CoreData.framework/Headers/CoreData.h:8:9: fatal error: could not build module 'Foundation'

  cargo:warning=#import <Foundation/Foundation.h>

  cargo:warning= ~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=While building module 'CoreData' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSKeyValueBinding.h:12:

  cargo:warning=While building module 'CloudKit' imported from /nix/store/904ymlq8dn6q5hi9yaq6z77628jis6r3-apple-framework-CoreData-11.0.0/Library/Frameworks/CoreData.framework/Headers/NSPersistentCloudKitContainerOptions.h:10:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=/nix/store/19lp09hc36s4aijcyyszjb09ammi5ill-apple-framework-CloudKit-11.0.0/Library/Frameworks/CloudKit.framework/Headers/CloudKit.h:8:9: fatal error: could not build module 'Foundation'

  cargo:warning=#import <Foundation/Foundation.h>

  cargo:warning= ~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=While building module 'CoreData' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSKeyValueBinding.h:12:

  cargo:warning=While building module 'CloudKit' imported from /nix/store/904ymlq8dn6q5hi9yaq6z77628jis6r3-apple-framework-CoreData-11.0.0/Library/Frameworks/CoreData.framework/Headers/NSPersistentCloudKitContainerOptions.h:10:

  cargo:warning=While building module 'CoreLocation' imported from /nix/store/19lp09hc36s4aijcyyszjb09ammi5ill-apple-framework-CloudKit-11.0.0/Library/Frameworks/CloudKit.framework/Headers/CKRecord.h:9:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/48ppfx34bv5nxzxj1wa0n521hj3nixsz-apple-framework-CoreLocation-11.0.0/Library/Frameworks/CoreLocation.framework/Headers/CoreLocation.h:18:

  cargo:warning=/nix/store/48ppfx34bv5nxzxj1wa0n521hj3nixsz-apple-framework-CoreLocation-11.0.0/Library/Frameworks/CoreLocation.framework/Headers/CLErrorDomain.h:10:9: fatal error: could not build module 'Foundation'

  cargo:warning=#import <Foundation/Foundation.h>

  cargo:warning= ~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=While building module 'CoreImage' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSColor.h:51:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CoreImage.h:8:

  cargo:warning=/nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CoreImageDefines.h:19:10: fatal error: could not build module 'ApplicationServices'

  cargo:warning=#include <ApplicationServices/ApplicationServices.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=While building module 'CoreImage' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSColor.h:51:

  cargo:warning=While building module 'CoreVideo' imported from /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CIImage.h:10:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CoreVideo.h:20:

  cargo:warning=In file included from /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CVReturn.h:21:

  cargo:warning=/nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CVBase.h:28:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CFBase.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=While building module 'CoreImage' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSColor.h:51:

  cargo:warning=While building module 'CoreVideo' imported from /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CIImage.h:10:

  cargo:warning=While building module 'IOSurface' imported from /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CVPixelBufferIOSurface.h:31:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/d8an1vz0bmni8fz7smvwawi2wksbaa2g-apple-framework-IOSurface-11.0.0/Library/Frameworks/IOSurface.framework/Headers/IOSurface.h:12:

  cargo:warning=/nix/store/d8an1vz0bmni8fz7smvwawi2wksbaa2g-apple-framework-IOSurface-11.0.0/Library/Frameworks/IOSurface.framework/Headers/IOSurfaceBase.h:36:10: fatal error: could not build module 'CoreFoundation'

  cargo:warning=#include <CoreFoundation/CoreFoundation.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=While building module 'CoreImage' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSColor.h:51:

  cargo:warning=While building module 'CoreVideo' imported from /nix/store/yw1h43xr5kbhv0yiz72nsa1a4pk16015-apple-framework-CoreImage-11.0.0/Library/Frameworks/CoreImage.framework/Headers/CIImage.h:10:

  cargo:warning=While building module 'Metal' imported from /nix/store/9iwrsrqrpsnms8801pbkjz1y0hmpfx82-apple-framework-CoreVideo-11.0.0/Library/Frameworks/CoreVideo.framework/Headers/CVMetalTextureCache.h:41:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/lbxzwnmn6lmmwzbdzlhw7fyq102ldqyh-apple-framework-Metal-11.0.0/Library/Frameworks/Metal.framework/Headers/Metal.h:9:

  cargo:warning=/nix/store/lbxzwnmn6lmmwzbdzlhw7fyq102ldqyh-apple-framework-Metal-11.0.0/Library/Frameworks/Metal.framework/Headers/MTLTypes.h:8:9: fatal error: could not build module 'Foundation'

  cargo:warning=#import <Foundation/Foundation.h>

  cargo:warning= ~~~~~~~^

  cargo:warning=While building module 'Cocoa' imported from objc/notify.h:1:

  cargo:warning=While building module 'AppKit' imported from /nix/store/2j5ymd7gxsd67xp7s9bfn1dss4apca7c-apple-framework-Cocoa-11.0.0/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:

  cargo:warning=While building module 'QuartzCore' imported from /nix/store/ws7r9sxk5fs5awpv8731iqgfinq48h8b-apple-framework-AppKit-11.0.0/Library/Frameworks/AppKit.framework/Headers/NSOpenGLLayer.h:11:

  cargo:warning=In file included from <module-includes>:1:

  cargo:warning=In file included from /nix/store/9699m64kz6cn19yfrypmcm7i72k25bhr-apple-framework-QuartzCore-11.0.0/Library/Frameworks/QuartzCore.framework/Headers/QuartzCore.h:9:

  cargo:warning=/nix/store/9699m64kz6cn19yfrypmcm7i72k25bhr-apple-framework-QuartzCore-11.0.0/Library/Frameworks/QuartzCore.framework/Headers/CoreImage.h:1:10: fatal error: could not build module 'CoreImage'

  cargo:warning=#include <CoreImage/CoreImage.h>

  cargo:warning= ~~~~~~~~^

  cargo:warning=In file included from objc/notify.m:1:

  cargo:warning=objc/notify.h:1:9: fatal error: could not build module 'Cocoa'

  cargo:warning=#import <Cocoa/Cocoa.h>

  cargo:warning= ~~~~~~~^

  cargo:warning=25 errors generated.

  exit status: 1

  --- stderr


  error occurred: Command "/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-arch" "x86_64" "-Wall" "-Wextra" "-fmodules" "-Wno-deprecated-declarations" "-mmacos-version-min=10.12" "-o" "/private/tmp/nix-build-cargo-watch-8.4.1.drv-0/source/target/x86_64-apple-darwin/release/build/mac-notification-sys-d4149a46f10e6591/out/objc/notify.o" "-c" "objc/notify.m" with args "cc" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
```

### Additional context

Sounds similar to this issue: https://discourse.nixos.org/t/need-help-from-darwin-users-syntax-errors-in-library-frameworks-foundation-framework-headers/30467

Though I have a hard time seeing any dependency that would be built with a different version of the framework in its closure. 

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@xrelkd @ivan @matthiasbeyer

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.0`
 - nixpkgs: `/nix/store/4h9p842jkjfmj2sm5i7s6pbnachinay7-source`
```

My system is showing as "aarch64-darwin", but I'm building with Rosetta. Hydra reports the same issue: https://hydra.nixos.org/build/241969194/nixlog/3

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/tools/rust/cargo-watch/default.nix
+++ b/pkgs/development/tools/rust/cargo-watch/default.nix
@@ -3,9 +3,7 @@
 , rustPlatform
 , fetchFromGitHub
 , Cocoa
-, CoreServices
 , Foundation
-, libiconv
 }:
 
 rustPlatform.buildRustPackage rec {
@@ -21,7 +19,9 @@ rustPlatform.buildRustPackage rec {
 
   cargoHash = "sha256-0D+aM/zap5UDQ+k9c/p+ZfN1OUjDzFRArvcmqEOcBbM=";
 
-  buildInputs = lib.optionals stdenv.isDarwin [ Cocoa CoreServices Foundation libiconv ];
+  buildInputs = lib.optionals stdenv.isDarwin [ Foundation Cocoa ];
+
+  NIX_LDFLAGS = lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ "-framework" "AppKit" ];
 
   # `test with_cargo` tries to call cargo-watch as a cargo subcommand
   # (calling cargo-watch with command `cargo watch`)

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -17149,8 +17149,8 @@ with pkgs;
   cargo-wasi = callPackage ../development/tools/rust/cargo-wasi {
     inherit (darwin.apple_sdk.frameworks) Security;
   };
-  cargo-watch = darwin.apple_sdk_11_0.callPackage ../development/tools/rust/cargo-watch {
-    inherit (darwin.apple_sdk_11_0.frameworks) Cocoa CoreServices Foundation;
+  cargo-watch = callPackage ../development/tools/rust/cargo-watch {
+    inherit (darwin.apple_sdk.frameworks) Foundation Cocoa;
   };
   cargo-wipe = callPackage ../development/tools/rust/cargo-wipe { };
   cargo-workspaces = callPackage ../development/tools/rust/cargo-workspaces { };

